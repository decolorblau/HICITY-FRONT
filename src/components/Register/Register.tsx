import React, { useState, useEffect } from "react";
import {
  Text,
  View,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView,
  Alert,
} from "react-native";
import styles from "../Login.styles";
import useUser from "../../hooks/useUser";
import { useNavigation } from "@react-navigation/core";
import RoutesEnum from "../../navigation/routes";
import { RegisterScreenNavigationProp } from "../../types/navigation.types";
import { validateEmail } from "../../utils/validations";

const Register = () => {
  const navigation = useNavigation<RegisterScreenNavigationProp>();

  const initialUser = {
    name: "",
    email: "",
    password: "",
  };

  const [userData, setUserData] = useState(initialUser);
  const [buttonDisabled, setButtonDisabled] = useState(true);

  const { registerUser } = useUser();

  const changeUserData = (text: string, identify: string) => {
    setUserData({
      ...userData,
      [identify]: text,
    });
  };

  useEffect(() => {
    setButtonDisabled(
      userData.name === "" ||
        userData.email === "" ||
        userData.password.length < 7
    );
  }, [userData.name, userData.email, userData.password]);

  const onSubmit = () => {
    if (validateEmail(userData.email)) {
      const newUser = {
        name: userData.name,
        email: userData.email.toLowerCase(),
        password: userData.password,
      };
      registerUser(newUser);
      resetForm();
      navigation.navigate(RoutesEnum.login);
    } else {
      Alert.alert("email no válido");
    }
  };

  const resetForm = () => {
    setUserData(initialUser);
  };

  return (
    <KeyboardAvoidingView behavior="padding" enabled={true}>
      <View>
        <Text style={styles.title}>CREA TU CUENTA</Text>
        <View>
          <View>
            <Text style={styles.label}>NOMBRE</Text>
            <TextInput
              style={styles.input}
              value={userData.name}
              placeholder="Tu nombre de usuario"
              onChangeText={(data) => changeUserData(data, "name")}
              testID="name"
              maxLength={20}
              textContentType="name"
            />
          </View>
          <View>
            <Text style={styles.label}>EMAIL</Text>
            <TextInput
              style={styles.input}
              textContentType="emailAddress"
              value={userData.email}
              placeholder="Email"
              onChangeText={(data) => changeUserData(data, "email")}
              testID="email"
              accessibilityLabel="email"
              maxLength={32}
            />
          </View>
          <View>
            <Text style={styles.label} testID="password">
              CONTRASEÑA
            </Text>
            <TextInput
              style={styles.input}
              value={userData.password}
              placeholder="Contraseña "
              onChangeText={(data) => changeUserData(data, "password")}
              testID="password"
              secureTextEntry={true}
              maxLength={20}
              textContentType="password"
            />
          </View>
        </View>

        <Text style={styles.inputHelp}>Mínimo 8 carácters</Text>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            onPress={onSubmit}
            disabled={buttonDisabled}
            style={buttonDisabled ? styles.buttonDisabled : styles.button}
          >
            <Text style={styles.buttonText}>COMENZAR</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity
            onPress={() => {
              navigation.navigate(RoutesEnum.login);
            }}
          >
            <Text>- LOGIN -</Text>
          </TouchableOpacity>
        </View>
      </View>
    </KeyboardAvoidingView>
  );
};

export default Register;
