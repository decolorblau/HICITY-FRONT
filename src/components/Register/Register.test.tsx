import React from "react";
import { Provider } from "react-redux";
import Register from "./Register";
import { render } from "@testing-library/react-native";
import mockStore from "../../../__mocks__/mockStore";
import { NavigationContainer } from "@react-navigation/native";

describe("Given a Register component", () => {
  describe("When it is rendered", () => {
    test("Then screen should match the snapshot", () => {
      const store = mockStore({ count: 5 });

      const screen = render(
        <Provider store={store}>
          <NavigationContainer>
            <Register />
          </NavigationContainer>
        </Provider>
      );

      expect(screen).toMatchSnapshot();
    });
  });
});
