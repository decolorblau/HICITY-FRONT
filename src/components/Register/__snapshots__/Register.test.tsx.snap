// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register component When it is rendered Then screen should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#ffcc00",
          "fontSize": 24,
          "fontWeight": "600",
          "marginBottom": 5,
        }
      }
    >
      CREA TU CUENTA
    </Text>
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#444444",
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          NOMBRE
        </Text>
        <TextInput
          allowFontScaling={true}
          maxLength={20}
          onChangeText={[Function]}
          placeholder="Tu nombre de usuario"
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomColor": "#888888",
              "borderColor": "#fff",
              "borderWidth": 1,
              "color": "#888888",
              "fontSize": 16,
              "height": 40,
              "margin": 0,
              "padding": 10,
              "paddingHorizontal": 0,
              "width": 300,
            }
          }
          testID="name"
          textContentType="name"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#444444",
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          EMAIL
        </Text>
        <TextInput
          accessibilityLabel="email"
          allowFontScaling={true}
          maxLength={32}
          onChangeText={[Function]}
          placeholder="Email"
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomColor": "#888888",
              "borderColor": "#fff",
              "borderWidth": 1,
              "color": "#888888",
              "fontSize": 16,
              "height": 40,
              "margin": 0,
              "padding": 10,
              "paddingHorizontal": 0,
              "width": 300,
            }
          }
          testID="email"
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#444444",
              "fontSize": 16,
              "marginTop": 15,
            }
          }
          testID="password"
        >
          CONTRASEÑA
        </Text>
        <TextInput
          allowFontScaling={true}
          maxLength={20}
          onChangeText={[Function]}
          placeholder="Contraseña "
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderBottomColor": "#888888",
              "borderColor": "#fff",
              "borderWidth": 1,
              "color": "#888888",
              "fontSize": 16,
              "height": 40,
              "margin": 0,
              "padding": 10,
              "paddingHorizontal": 0,
              "width": 300,
            }
          }
          testID="password"
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#888888",
          "fontSize": 14,
          "marginTop": 5,
        }
      }
    >
      Mínimo 8 carácters
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeca1",
            "borderRadius": 90,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "padding": 10,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          COMENZAR
        </Text>
      </View>
    </View>
  </View>
</View>
`;
